gnd/src/main/java/com/google/android/gnd/ui/projectselector/ProjectSelectorViewModel.java

```diff
@@ -15,48 +15,80 @@
  */
 package com.google.android.gnd.ui.projectselector;
 
+import android.util.Log;
 import androidx.lifecycle.LiveData;
 import androidx.lifecycle.MutableLiveData;
 import com.google.android.gnd.repository.DataRepository;
 import com.google.android.gnd.repository.Resource;
 import com.google.android.gnd.system.AuthenticationManager;
 import com.google.android.gnd.ui.common.AbstractViewModel;
 import com.google.android.gnd.vo.Project;
-import io.reactivex.Completable;
+import io.reactivex.Single;
+import io.reactivex.subjects.PublishSubject;
 import java.util.Collections;
 import java.util.List;
 import javax.inject.Inject;
 
 public class ProjectSelectorViewModel extends AbstractViewModel {
   private static final String TAG = ProjectSelectorViewModel.class.getSimpleName();
 
-  private final DataRepository dataRepository;
   private final MutableLiveData<Resource<List<Project>>> projectSummaries;
-  private final AuthenticationManager authManager;
+  private final PublishSubject<Integer> projectSelections;
+  private final MutableLiveData<Project> activeProject;
+  private final MutableLiveData<Throwable> activateProjectErrors;
 
   @Inject
   ProjectSelectorViewModel(DataRepository dataRepository, AuthenticationManager authManager) {
-    this.dataRepository = dataRepository;
     this.projectSummaries = new MutableLiveData<>();
-    this.authManager = authManager;
-  }
+    this.activeProject = new MutableLiveData<>();
+    this.activateProjectErrors = new MutableLiveData<>();
+    this.projectSelections = PublishSubject.create();
+
+    disposeOnClear(
+        projectSelections
+            .switchMapSingle(
+                idx ->
+                    dataRepository
+                        .activateProject(getProjectSummary(idx).getId())
+                        .doOnError(this::onActiveProjectError)
+                        .onErrorResumeNext(Single.never()))
+            .subscribe(activeProject::setValue));
+
+    AuthenticationManager.User user =
+        authManager.getUser().blockingFirst(AuthenticationManager.User.ANONYMOUS);
```

Comment:
Note to self: We should add a TODO for this and the other instance of `blockingFirst` to re-implement this in the `AuthenticationManager`.